cabal-install/src/Distribution/Client/Init/Prompt.hs

```diff
@@ -143,14 +144,14 @@ promptDefault
       -- ^ pretty-printer
     -> String
       -- ^ prompt message
-    -> Maybe t
+    -> (Either Bool t)
```

Comment:
Given that the type and its constructors appear with this meaning in so many places, would you like to define a custom datatype `InitValueDefault t | InitValueMandatory | InitValueOptional` (but with better names)? Boolean blindness is one thing, but here it's more complex and I quickly forget which constructor means what. By convention `Left` is an error, but this convention doesn't help here and I think there are no other conventions that would help.

Reply:
Actually, should this be `InitValueDefault !t`? I'm so used to `StrictData` (which we don't use in cabal for some reason), that I forgot the trade-offs.

Reply:
Here we go, custom datatyped.

Reply:
Thank you!